### PR TITLE
Fixed: Duplicate emails and comments

### PR DESCRIPTION
**Pull request by chaikovskyia**

### CODE REVIEW

File: Model/Dintero.php [hunk 1]
```diff
@@ -291,6 +292,11 @@ public function process($merchantOrderId, $transactionId, $sessionId = null)
 
         $this->getResponse()->setData($this->client->getTransaction($transactionId, $order->getStoreId()));
 
+        // if payment has transaction that means that it was already processed
+        if ($payment->getLastTransId()) {
```
Comment (andersem) [inline Model/Dintero.php:296]: Might be outdated If it happens in the same second?

File: Model/Dintero.php [hunk 2]
```diff
@@ -335,7 +341,7 @@ public function processOrder($order)
         $this->addStatusComment($payment);
         $order->save();
 
-        if (!$isFailed) {
+        if (!$isFailed && !$order->getEmailSent()) {
```
Comment (andersem) [inline Model/Dintero.php:344]: Might be outdated If it happens in the same second?

**Review by andersem — APPROVED** [review]
Looks fine overall, but what if the other source of processing the order updates it in the same second. Won't it still be possible that we get race conditions?

File: Model/Dintero.php (hunk 2 above)
Comment (chaikovskyia) [reply Model/Dintero.php]: @andersem , I've added additional check for this case.

**Review by andersem — APPROVED** [review]

**Pull request merged**